h2o-automl/src/main/java/ai/h2o/automl/modeling/StackedEnsembleStepsProvider.java

```diff
@@ -130,6 +139,59 @@ protected boolean isStackedEnsemble(Key<Model> key) {
                     }
                 },
+                new StackedEnsembleModelStep("monotone", DEFAULT_MODEL_TRAINING_WEIGHT, aml()) {
+                    { _description = _description+" (built using monotonically constrained AutoML models)"; }
+
+                    boolean hasMonotoneConstrains(Key<Model> modelKey) {
+                        Model model = DKV.getGet(modelKey);
+                        try {
+                            KeyValue[] mc = (KeyValue[]) PojoUtils.getFieldValue(
+                                    model._parms, "_monotone_constraints",
+                                    PojoUtils.FieldNaming.CONSISTENT);
+                            return mc != null && mc.length > 0;
+                        } catch (IllegalArgumentException e) {
+                            return false;
+                        }
+                    }
+
+                    @Override
+                    protected boolean canRun() {
+                        boolean canRun = super.canRun();
+                        if (!canRun) return false;
+                        int monotoneModels=0;
+                        for (Key<Model> modelKey: getTrainedModelsKeys()) {
+                            if (hasMonotoneConstrains(modelKey))
+                                monotoneModels++;
+                            if (monotoneModels >= 2)
+                                return true;
+                        }
+                        if (monotoneModels == 1) {
+                            aml().job().update(getAllocatedWork().consume(),
+                                    "Only one monotone base model; skipping this StackedEnsemble");
+                            aml().eventLog().info(EventLogEntry.Stage.ModelTraining,
+                                    String.format("Skipping StackedEnsemble '%s' since there is only one monotone model to stack", _id));
+                        } else {
+                            aml().job().update(getAllocatedWork().consume(),
+                                    "No monotone base model; skipping this StackedEnsemble");
+                            aml().eventLog().info(EventLogEntry.Stage.ModelTraining,
+                                    String.format("Skipping StackedEnsemble '%s' since there is no monotone model to stack", _id));
+                        }
+                        return false;
+                    }
+
+                    @Override
+                    @SuppressWarnings("unchecked")
+                    protected Key<Model>[] getBaseModels() {
+                        return Stream.of(getTrainedModelsKeys())
+                                .filter(k -> !isStackedEnsemble(k) && hasMonotoneConstrains(k))
+                                .toArray(Key[]::new);
+                    }
+
+                    @Override
+                    protected Job<StackedEnsembleModel> startJob() {
+                        return stack(_algo + "_MonotonicallyConstrainedModels", getBaseModels(), true);
```

Comment:
I know this has been merged but I didn't get a chance to review... @tomasfryda @sebhrusen Do you think it might be better to use a shorter name? e.g. `"_Monotonic"` or something?  Our Model ID names, especially SE, is already quite long and making the first column in the leaderboard very wide.

Reply:
I agree, "Monotonic" is better.

Reply:
I created a new PR since I failed to include it into this one: https://github.com/h2oai/h2o-3/pull/4656